### ISSUE TITLE
Null Pointer accessing WeakReference after update to 1.10.7

### ISSUE THREAD

**Issue by ssuchanowski** [BranchMetrics/Android-Deferred-Deep-Linking-SDK#217]
```
java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.Object java.lang.ref.WeakReference.get()' on a null object reference
                                                                              at io.branch.referral.Branch.handleNewRequest(Branch.java:2946)
                                                                              at io.branch.referral.Branch.userCompletedAction(Branch.java:1580)
```

**Comment by sojanpr** [comment]
@ssuchanowski  Thanks for reporting the issue. Sorry for this issue which is really caused by a memory leak fix. Will fix this asap and update

**Comment by sojanpr** [comment]
@ssuchanowski  We could reproduce the  issue. Issue is caused by a null pointer when calling `userActionCompleted` before branch get initialized. We open a new PR https://github.com/BranchMetrics/Android-Deferred-Deep-Linking-SDK/pull/218 for fixing this issue. Will release this after the testing.
Thanks

**Comment by sojanpr** [comment]
@ssuchanowski  We have released v1.10.8 which fix this issue. 
Thank you.
